index.js

```diff
@@ -1,21 +1,132 @@
 'use strict';
 const {app, BrowserWindow} = require('electron');
 
-module.exports = async (url, {type = 'json', ...options} = {}) => {
+let mainWindow;
+let isReady = false;
+
+module.exports = async (url, options = {}) => {
 	await app.whenReady();
 
-	const mainWindow = new BrowserWindow({
-		show: false,
-		sandbox: true
+	if (!mainWindow) {
+		mainWindow = new BrowserWindow({
+			show: false,
+			sandbox: true
+		});
+
+		mainWindow.loadURL('about:blank');
+		await mainWindow.webContents.executeJavaScript(ProxyFetch.toString());
+		await mainWindow.webContents.executeJavaScript('window.fetcher = new ProxyFetch();');
+		isReady = true;
+	}
+
+	while (!isReady) {
+		await sleep(30);
+	}
+
+	return ProxyFetch.unproxy(await mainWindow.webContents.executeJavaScript(`
+		window.fetcher.fetch('${url}', JSON.parse('${JSON.stringify(options)}'));
+	`, true));
+};
+
+class ProxyFetch {
+	constructor() {
+		this.autoincrement = 0;
+		this.responses = {};
+	}
+
+	async fetch(...args) {
+		const response = await fetch(...args);
+		const id = this.autoincrement++;
+		this.responses[id] = response;
+		return this.newResponse(id, response);
+	}
+
+	newResponse(id, response) {
+		const keys = [];
+		const entries = [];
+		const headers = {};
+		const isFunc = {isFunc: true};
+
+		for (let [key, value] of response.headers.entries()) {
+			keys.push(key = key.toLowerCase());
+			entries.push([key, value]);
+			headers[key] = headers[key] ? `${headers[key]},${value}` : value;
+		}
+
+		return {
+			id,
+			type: response.type,
+			url: response.url,
+			redirected: response.redirected,
+			status: response.status,
+			ok: response.ok,
+			statusText: response.statusText,
+			bodyUsed: response.bodyUsed,
+			headers: {
+				headers,
+				keys,
+				entries
+			},
+			blob: isFunc,
+			json: isFunc,
+			text: isFunc
+		};
+	}
+
+	call(id, method, ...args) {
+		const response = this.responses[id];
+		if (!response) {
+			throw new Error('Could not find response');
+		}
+
+		delete this.responses[id];
+		return response[method](...args);
+	}
+}
+
+ProxyFetch.unproxy = function (response) {
+	const {keys, entries, headers} = response.headers;
+
+	response.headers = new FetchHeaders({
+		keys: () => keys,
+		entries: () => entries,
+		get: n => headers[n.toLowerCase()],
+		has: n => n.toLowerCase() in headers
 	});
 
-	await mainWindow.loadURL('about:blank');
+	for (const k in response) {
```

Comment:
Use `for-of` loop.

Reply:
I think it's not possible in that specific case, we need to overwrite the response object therefore it needs the `key`.

Reply:
Of course it's possible. `for (const k in Object.keys(response)) {}`